pyresample/geometry.py

```diff
@@ -958,21 +967,39 @@ def freeze(self, lonslats=None, resolution=None, shape=None, proj_info=None):
         shape = None if None in shape else shape
         area_extent = self.area_extent
         if not area_extent or not width or not height:
-            proj4 = Proj(proj_dict)
-            try:
-                lons, lats = lonslats
-            except (TypeError, ValueError):
-                lons, lats = lonslats.get_lonlats()
-            xarr, yarr = proj4(np.asarray(lons), np.asarray(lats))
-            xarr[xarr > 9e29] = np.nan
-            yarr[yarr > 9e29] = np.nan
-            corners = [np.nanmin(xarr), np.nanmin(yarr),
-                       np.nanmax(xarr), np.nanmax(yarr)]
+            corners = self._compute_corners(proj_dict, lonslats)
             area_extent, width, height = self.compute_domain(corners, resolution, shape)
         return AreaDefinition(self.area_id, self.description, '',
                               projection, width, height,
                               area_extent, self.rotation)
 
+    def _compute_corners(self, proj_dict, lonslats):
```

Comment:
Could this be renamed to `compute_bounds`?
I think corners is misleading here, as we just return the x and y bounds.

Reply:
Hm I used corners because that was the name of the variable before and it is the name of the argument to `compute_domain`. I could change it to `_compute_bound_centers` or something?

Reply:
Go ahead!